media/js/firefox/onboarding/user-actions.js

```diff
@@ -0,0 +1,70 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+(function(Mozilla, $) {
+    'use strict';
+
+    var client = Mozilla.Client;
+
+    // *************
+    // pin it/search variation
+    // *************
+    var $trySearch = $('#search-now');
+    var $searchPointer = $('#search-pointer');
+
+    // as this is for firstrun, search should almost certainly be available
+    // best to double check though
+    function shouldPromoteSearch() {
+        var availableTargets = new Promise(function(resolve, reject) {
+            Mozilla.UITour.getConfiguration('availableTargets', function(config) {
+                if (config && config.targets) {
+                    resolve(config.targets);
+                } else {
+                    reject('UITour: targets property not found.');
+                }
+            });
+        });
+
+        return new Promise(function(resolve, reject) {
+            Promise.all([availableTargets]).then(function(results) {
+                resolve(results[0] && results[0].indexOf('search') > -1);
+            }, function(reason) {
+                reject(reason);
+            });
+        });
+    }
+
+    function openSearchUI(e) {
+        e.preventDefault();
+        Mozilla.UITour.openSearchPanel(function() {
+            Mozilla.UITour.setSearchTerm('What is the weather?');
+            $trySearch.addClass('hidden');
+            $searchPointer.removeClass('hidden');
+        });
+    }
+
+    function initSearchUI() {
+        shouldPromoteSearch().then(function() {
+            $trySearch.removeClass('hidden');
+            $trySearch.on('click', openSearchUI);
+        });
+    }
+
+    // only 45+ should hit this page, but safer to double check
+    if (client.isFirefoxDesktop && client.FirefoxMajorVersion >= 43) {
```

Comment:
I'm not sure if this check for `43` is necessary? We needed it on the Yahoo page since we also had `getConfiguration('search')`, but that's not being used here it seems. `getConfiguration('availableTargets')` should work for any version of Firefox that supports UITour. If you need some kind of simple check that UITour is working, a [ping](https://github.com/mozilla/bedrock/blob/master/media/js/base/uitour-lib.js#L245) callback might be enough, although I think the check for `isFirefoxDesktop` should be kept.